DataFormats/L1Trigger/interface/P2GTCandidate.h

```diff
@@ -0,0 +1,328 @@
+#ifndef DataFormats_L1Trigger_P2GTCandidate_h
+#define DataFormats_L1Trigger_P2GTCandidate_h
+
+#include <vector>
+#include <ap_int.h>
+#include <stdexcept>
+
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/Common/interface/RefVector.h"
+
+namespace l1t {
+
+  class L1GTProducer;
+
+  class P2GTCandidate;
+  typedef std::vector<P2GTCandidate> P2GTCandidateCollection;
+  typedef edm::Ref<P2GTCandidateCollection> P2GTCandidateRef;
+  typedef edm::RefVector<P2GTCandidateCollection> P2GTCandidateRefVector;
+  typedef std::vector<P2GTCandidateRef> P2GTCandidateVectorRef;
+
+  class P2GTCandidate {
```

Comment:
Why doesn't P2GTCandidate derive from the Candidate class?
That would simplify things a lot!

Reply:
I think we (from the GT side)  have a better understanding on what the requirements for HLT are now. We will address the issues internally and come back to you next week.

Reply:
@Martin-Grunewald
> Why doesn't P2GTCandidate derive from a class within the `Candidate` class hierarchy?

We initially thought we wouldn't need the physical values provided by `Candidate`. We will change it to be a subclass of `Candidate`, however to the get the precise type (e.g. not only `Muon` but `TkMuon`) you would still have to cast it to a `P2GTCandidate`. So I'm not sure how much this would help.

> Why is there a single C++ type for all physics types (track, electron, muon, ...)? The first thing people need to check is physics type, no?

This is by design to have our conditions as generic as possible (analogously to our FW implementation) and to have control over the scales to ensure consistency. An alternative implementation would be to template the conditions but we decided against this.

Reply:
I do not understand the your about having to cast it to a P2GTCandidate - this type is there regardless of whether P2GTCandidate type itself is subclassed in the Candidate hierarchy or not. ??

Reply:
@Martin-Grunewald my understanding of having it be a subclass of `Candidate` is that one would get it as a `Candidate` from the event or use it as a `Candidate` in a function. If that's the case then to get the precise particle type one would need to `dynamic_cast` it to an `P2GTCandidate` to get this info anyways.

Reply:
Well, at HLT we would get it as a collection of P2GTCandiates and use some of the methods of/from the Candidate hierarchy on each object of the collection (eg, the physics 4-momentum), but also accessing some P2GTCandiate specific info (such as physics type Muon vs TkMuon etc.)